### PR TITLE
AMP-99103 [DatabricksImport] Update Databricks Import developer doc for GA

### PR DESCRIPTION
**Pull request by bh202205**
# Amplitude Developer Docs PR

## Description

Updated the developer doc for databricks import for GA

## Deadline

When do these changes need to be live on the site?

Monday, May 6th, 2024

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

**Review by markzegarelli — APPROVED** [review]

File: docs/data/sources/databricks.md [hunk 1]
```diff
@@ -39,19 +37,48 @@ Ensure that the new cluster can run jobs by NOT having configs below in cluster'
 }
 ```
 
+Ensure that the your cluster has python version >= 3.9; Otherwise, you may see the following error in your workflow job:
+
+```json
+TypeError: 'type' object is not subscriptable
+```
+
+#### Policies and Access Modes
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:46]: would be good to rename title to `Cluster Policies and Access Modes` to make it more cluster specific, so customer won't confuse this with sections of configuring policies and permissions for PATs.

File: docs/data/sources/databricks.md [hunk 2]
```diff
@@ -39,19 +37,48 @@ Ensure that the new cluster can run jobs by NOT having configs below in cluster'
...
+
+Amplitude supports all policies and access modes. However, if your clusters have the following policies and access modes, grant the Data Reader permission (`USE CATALOG`, `USE SCHEMA`, `EXECUTE`, `READ` `VOLUME`, `SELECT`) to the your workspace user or service principal, whose personal access token is used to authenticate in Amplitude. Otherwise, you can't  access the tables in the unity catalog of your import source.
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:48]: nit: `READ` `VOLUME` shall be `READ VOLUME`?

File: docs/data/sources/databricks.md [hunk 3]
```diff
@@ -39,19 +37,48 @@ Ensure that the new cluster can run jobs by NOT having configs below in cluster'
...
+
+AWS Databricks:
+
+- Unrestricted - Multi node - No isolation shared
+- Unrestricted - Single node - No isolation shared
+- Power User Compute - No isolation shared
+- Legacy Shared Compute
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:55]: Felt like turn these combinations into table format will be easier to understand, for example

```
| Policy | Node | Cluster Access mode | 
--------------------------------------
| Unrestricted | Multi node | No isolation shared|
....
```

File: docs/data/sources/databricks.md [hunk 4]
```diff
@@ -39,19 +37,48 @@ Ensure that the new cluster can run jobs by NOT having configs below in cluster'
...
+
+GCP Databricks:
+
+- Unrestricted - Multi node - No isolation shared
+- Unrestricted - Single node - No isolation shared
+- Power User Compute - Shared
+- Power User Compute - No isolation shared
+- Legacy Shared Compute
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:63]: same as above

File: docs/data/sources/databricks.md [hunk 5]
```diff
@@ -30,6 +26,8 @@ Before you start to configure the Databricks source in Amplitude, complete the f
 
 Amplitude creates workflows in this cluster on your behalf to start sync jobs. When complete, copy the **Server hostname** and **Http path** values to use in a later step. Find both values on the **Configuration -> JDBC/ODBC** tab. For more information about cluster types, see [Compute](https://docs.databricks.com/en/compute/index.html).
 
+![a screenshot indicating where to find server host name and Http path](../../assets/images/integrations-databricks-import-server-hostname-http-path.png)
+
 Ensure that the new cluster can run jobs by NOT having configs below in cluster's policy. See details in Databricks' article [Policy definition](https://docs.databricks.com/en/administration-guide/clusters/policy-definition.html#workload).
 
 ```json
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:33]: Can we use to mark this as important notes? Here is what i found https://github.com/orgs/community/discussions/16925

```
> [!IMPORTANT]  
> Crucial information necessary for users to succeed.
```

File: docs/data/sources/databricks.md [hunk 6]
```diff
@@ -39,19 +37,48 @@ Ensure that the new cluster can run jobs by NOT having configs below in cluster'
 }
 ```
 
+Ensure that the your cluster has python version >= 3.9; Otherwise, you may see the following error in your workflow job:
+
+```json
+TypeError: 'type' object is not subscriptable
+```
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:44]: Same to mark this as important notes?

File: docs/data/sources/databricks.md [hunk 7]
```diff
@@ -84,26 +106,28 @@ To add Databricks as a source in Amplitude, complete the following steps.
...
 4. Click **Next** to verify access.
 
 ### Select data to import
 
-1. Select the Databricks table(s) from which Amplitude pulls data.
+1. Select the Databricks tables from which Amplitude pulls data.
 2. Select the version of the table(s) that Amplitude imports from. Select **First** or **Latest**.
-3. Select the data type to import from the selected table(s). The Databricks source supports three data types.
+First:  
+Latest:
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:119]: seems like explaination is not added? `First` means first version, which is 0. `Latest` means latest version.

File: docs/data/sources/databricks.md [hunk 8]
```diff
@@ -84,26 +106,28 @@ To add Databricks as a source in Amplitude, complete the following steps.
...
+    - Server hostname
+    - Http Path
     - Personal Access Token (for the workspace user or Service Principal)
 4. Click **Next** to verify access.
 
 ### Select data to import
 
-1. Select the Databricks table(s) from which Amplitude pulls data.
+1. Select the Databricks tables from which Amplitude pulls data.
 2. Select the version of the table(s) that Amplitude imports from. Select **First** or **Latest**.
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:117]: maybe rephrase this a bit `Select the version of the table(s) that Amplitude imports from` -> `Select the table version for initial import. Initial import will import everything from table as of the selected version.`

File: docs/data/sources/databricks.md [hunk 9]
```diff
@@ -84,26 +106,28 @@ To add Databricks as a source in Amplitude, complete the following steps.
...
-3. Select the data type to import from the selected table(s). The Databricks source supports three data types.
+First:  
+Latest: 
+3. Select the data type to import from the selected tables. The Databricks source supports three data types.
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:120]: nit: probably rephrase `Select the data type to import from the selected tables.` -> `Select the data type for data to be imported`

File: docs/data/sources/databricks.md [hunk 10]
```diff
@@ -84,26 +106,28 @@ To add Databricks as a source in Amplitude, complete the following steps.
...
     - Event
     - User properties
     - Group properties
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:123]: Can we add link for each of the data type to help them understand what is that?
`Event`: https://www.docs.developers.amplitude.com/analytics/what-is-amplitude/#events
`User properties`: https://www.docs.developers.amplitude.com/analytics/what-is-amplitude/#user-properties
`Group properties`: https://help.amplitude.com/hc/en-us/articles/115001765532-Account-level-reporting-in-Amplitude#account-level-properties-group-properties

File: docs/data/sources/databricks.md [hunk 11]
```diff
@@ -84,26 +106,28 @@ To add Databricks as a source in Amplitude, complete the following steps.
...
+    For the **Event** data type, optionally select **Sync User Properties** or **Sync Group Properties** to sync the corresponding properties *within* an event.
 
 4. Set the sync frequency. This frequency determines the interval at which Amplitude pulls data from Databricks.
 5. Configure the SQL command that transforms data in Databricks before Amplitude imports it.
-    - Amplitude treats each record in the SQL execution as an event to be import. See the Example body in the [Batch Event Upload API](/analytics/apis/batch-event-upload-api/#example-body) documentation to ensure each record you import complies.
-    - Amplitude can transform / import from only the tables you specify in step 1 above.
+    - Amplitude treats each record in the SQL execution output as an event to be import. See the Example body in the [Batch Event Upload API](/analytics/apis/batch-event-upload-api/#example-body) documentation to ensure each record you import complies.
+    - Amplitude can transform / import from only the tables you specify in step 1 above. TODO: add in an example to explain it
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:130]: Add an example?

**Comment by bh202205** [comment]
@fzqgriffin Thank you for providing all these comments and I have addressed them all. Please take a look at the PR again when you have time. Thanks!

File: docs/data/sources/databricks.md [hunk 12]
```diff
@@ -84,26 +112,29 @@ To add Databricks as a source in Amplitude, complete the following steps.
...
 
 ### Select data to import
 
-1. Select the Databricks table(s) from which Amplitude pulls data.
-2. Select the version of the table(s) that Amplitude imports from. Select **First** or **Latest**.
-3. Select the data type to import from the selected table(s). The Databricks source supports three data types.
-    - Event
-    - User properties
-    - Group properties
+1. Select the Databricks tables from which Amplitude pulls data.
```
Comment (fzqgriffin) [inline docs/data/sources/databricks.md:122]: Actually we don't have step 1 anymore since all tables are automatically selected when configuring SQL. So a good ordering for steps shall probably be
1. Configure SQL command
2. Select data type.
3. TestSQL
4. Select table version for initial import.
5. set sync frequency
6. then save 

By doing that, we can remove line 137 since table selected will always be the one used in SQL.

Comment (fzqgriffin) [reply docs/data/sources/databricks.md:122]: @bh202205 i have updated the doc, enjoy your vacation and don't need to worry about this.

**Review by fzqgriffin — APPROVED** [review]
